app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy like dislike]
+
+  def index
+    @posts = Post.includes(:likes).all
```

Comment:
@nilp06 
1. there is no need to add .all at the end.
2. Include comments also to avoid N+1 issue